src/main/kotlin/com/dsrm/dsrmbackend/validators/UserRolesValidator.kt

```diff
@@ -0,0 +1,37 @@
+package com.dsrm.dsrmbackend.validators
+
+import com.dsrm.dsrmbackend.repositories.RoleRepo
+import org.hibernate.validator.constraintvalidation.HibernateConstraintValidatorContext
+import org.springframework.beans.factory.annotation.Autowired
+import javax.validation.ConstraintValidator
+import javax.validation.ConstraintValidatorContext
+
+class UserRolesValidator : ConstraintValidator<UserRolesConstraint, List<String>> {
+
+    @Autowired
+    lateinit var roleRepo : RoleRepo
+
+    override fun isValid(userRoles: List<String>?, context: ConstraintValidatorContext?): Boolean {
+        if (userRoles == null) {
+            return true
+        }
+        val invalidRoles: MutableList<String> = mutableListOf()
+        for (roleId: String in userRoles) {
+            val maybeRole = roleRepo.findById(roleId)
```

Comment:
I would fetch all roles. Create method ```findByIdIn``` and pass whole list. And later remove all which exists.